Sources/MapboxCoreNavigation/NavigationEventsManager.swift

```diff
@@ -147,13 +153,17 @@ open class NavigationEventsManager {
         return eventDictionary
     }
     
-    func navigationFeedbackEvent(type: FeedbackType, description: String?) -> NavigationEventDetails? {
+    func navigationFeedbackEvent(type: FeedbackType, description: String?, withAppMetadata: [String:String?]? = nil) -> NavigationEventDetails? {
         guard let dataSource = dataSource, let sessionState = sessionState else { return nil }
-
-        var event = NavigationEventDetails(dataSource: dataSource, session: sessionState, defaultInterface: usesDefaultUserInterface)
+        if (withAppMetadata != nil) {
+            appMetadata = withAppMetadata
```

Comment:
Is the intent that passing in metadata would update the cached metadata but omitting it would reuse whatever had been passed in before? If so, I think it would be clearer to move this statement to the call sites.